lib/AST/ASTScopeCreation.cpp

```diff
@@ -209,8 +209,8 @@ void ASTScope::
 }
 
 void ASTScope::expandFunctionBody(AbstractFunctionDecl *AFD) {
-  auto *const SF = AFD->getParentSourceFile();
```

Comment:
I think the problem here is that we've never actually had a C++ function with a body before. And because there's no source file associated with a C++ decl context, it makes sense that this might be null.